git_machete/client.py

```diff
@@ -602,8 +602,17 @@ def connected_with_green_edge(bd: LocalBranchShortName) -> bool:
             else:
                 return
 
-        ans = self.ask_if(f"\nBranch {bold(down_branch)} is now merged into {bold(branch)}. Slide {bold(down_branch)} out of the tree of branch dependencies?" + get_pretty_choices('y', 'N'),
-                          f"\nBranch {bold(down_branch)} is now merged into {bold(branch)}. Sliding {bold(down_branch)} out of the tree of branch dependencies...", opt_yes=opt_yes)
+        remote = self.__git.get_strict_remote_for_fetching_of_branch(branch)
+        ans = self.ask_if(f"\nBranch {bold(branch)} is now fast-forwarded to match {bold(down_branch)}. Push {bold(branch)} to {bold(remote)}?" + get_pretty_choices('y', 'N'),
+                          f"\nBranch {bold(branch)} is now fast-forwarded to match {bold(down_branch)}. Pushing {bold(branch)} to {bold(remote)}...", opt_yes=opt_yes)
+        if ans in ('y', 'yes'):
+            self.__git.push(remote, branch)
+            branch_pushed_or_fast_forwarded_msg = f"\nBranch {bold(branch)} is now pushed to {bold(remote)}."
```

Comment:
```suggestion
            branch_pushed_or_fast_forwarded_msg = f"\nBranch {bold(branch)} is now fast-forwarded and pushed to {bold(remote)}."
```

Coz suggesting slide-out of `down_branch` only makes sense once `branch` is fast-forwarded to `down_branch`! Pushing `branch` is incidental/unrelated to sliding-out of `down_branch`.

Reply:
So we'd better mention the fast-forward as the reason for slide-out in both cases (push or no push)

Reply:
Awww okay I see now:

![image](https://user-images.githubusercontent.com/3383210/159664591-082d57db-f9b7-4459-93d4-b35677484c99.png)

So it's good! let's leave the current version